ansible/roles/debops.prosody/defaults/main.yml

```diff
@@ -0,0 +1,162 @@
+                           ansible_local.pki.realm|d())
+                       else "domain" }}'
+                                                                   # ]]]
+# .. envvar:: prosody__pki_crt_filename [[[
+#
+# Default CRT file name to use.
+prosody__pki_crt_filename: '{{ ansible_local.pki.crt
+                              if (ansible_local|d() and ansible_local.pki|d() and
+                                  ansible_local.pki.crt|d())
+                              else "default.crt" }}'
+                                                                   # ]]]
+# .. envvar:: prosody__pki_key_filename [[[
+#
+# Default private key file name to use.
+prosody__pki_key_filename: '{{ ansible_local.pki.key
+                              if (ansible_local|d() and ansible_local.pki|d() and
+                                  ansible_local.pki.key|d())
+                              else "default.key" }}'
+                                                                   # ]]]
+# .. envvar:: prosody_pki_hook_name [[[
+#
+# Name of the hook script which will be stored in hook directory.
+prosody_pki_hook_name: 'prosody'
+
+                                                                   # ]]]
+# .. envvar:: prosody_pki_hook_path [[[
+#
+# Directory with PKI hooks.
+prosody_pki_hook_path: '{{ ansible_local.pki.hooks
+                         if (ansible_local|d() and ansible_local.pki|d() and
+                             ansible_local.pki.hooks|d())
+                         else "/etc/pki/hooks" }}'
+
+                                                                   # ]]]
+# .. envvar:: prosody_pki_hook_action [[[
+#
+# Specify how changes in PKI should affect prosody, either 'reload' or 'restart'.
+prosody_pki_hook_action: 'reload'
+
+                                                                   # ]]]
+                                                                   # ]]]
+# .. envvar:: prosody__admins [[[
+#
+# list of admins jabber accounts of this prosody instance
+prosody__admins: []
+                                                                   # ]]]
+# .. envvar:: prosody__config_global: [[[
+#
+# main prosody configuration
+# @see prosody.im
+prosody__config_global:
+  admins: '{{ prosody__admins }}'
+  modules_enabled:
+    - "roster"
+    - "saslauth"
+    - "tls"
+    - "dialback"
+    - "disco"
+    - "private"
+    - "vcard"
+    - "privacy" #@TODO check version < 0.10
+    - "compression" #@TODO check lua-zlib
+    - "version"
+    - "uptime"
+    - "time"
+    - "ping"
+    - "pep"
+    - "admin_adhoc"
+    - "posix"
+    - "groups"
+    - "carbon"
+    - "mam"
+    - "blocking"
+    - "smacks"
+    #- "s2s_auth_dane"
+  allow_registration: False
+  daemonize: True
+  pidfile: "/var/run/prosody/prosody.pid"
+  #ssl:
+  #  key: '..'
+  #  certificate: '..'
+  c2s_require_encryption: True
+  s2s_secure_auth: True
+  s2s_insecure_domains:
+    - "gmail.com"
```

Comment:
This is an arbitrary configuration choice and might not be obvious for the first time user, therefore it should be avoided as a default. Instead you can add it as an example in the Getting Started documentation of the role, to inform the user that this might be needed for the role to work correctly.

DebOps is aimed for production use, and we should avoid an "insecure by default" approach whenever possible.

Reply:
Separate variable for insecure domains is good. You can explain the issue with `gmail.com` contacts in the "Getting started" section of the role documentation where hopefully users will look when they want to enable this role and encounter issues. Then they can decide if they want to enable insecure connections for `gmail.com` domain or not. This should at least ensure that they are aware of the risk - otherwise they would just enable the role, which works out of the box, but users don't know that connections to `gmail.com` are insecure.